test/formula.jl

```diff
@@ -1,134 +1,125 @@
 @testset "formula" begin
 
-    # TODO:
-    # - grouped variables in formulas with interactions
-    # - is it fast?  Can expand() handle DataFrames?
-    # - deal with intercepts
-    # - implement ^2 for datavector's
-    # - support more transformations with I()?
+    using StatsModels: hasresponse, hasintercept
 
-    ## Formula parsing
-    using StatsModels: @formula, Formula, Terms
+    y, x1, x2, x3, a, b, c, One = term(:y, :x1, :x2, :x3, :a, :b, :c, 1)
 
     ## totally empty
-    t = Terms(@eval @formula $(:($nothing ~ 0)))
-    @test t.response == false
-    @test t.intercept == false
-    @test t.terms == []
-    @test t.eterms == []
+    @test_broken t = @eval @formula $(:($nothing ~ 0))
+    @test_broken hasresponse(t) == false
+    @test_broken hasintercept(t) == false
+    @test_broken t.rhs == ConstantTerm(0)
+    @test_broken issetequal(terms(t), [ConstantTerm(0)])
 
     ## empty RHS
-    t = Terms(@formula(y ~ 0))
-    @test t.intercept == false
-    @test t.terms == []
-    @test t.eterms == [:y]
-    t = Terms(@formula(y ~ -1))
-    @test t.intercept == false
-    @test t.terms == []
+    t = @formula(y ~ 0)
+    @test hasintercept(t) == false
+    @test t.rhs == ConstantTerm(0)
+    @test issetequal(terms(t), term(:y, 0))
+    t = @formula(y ~ -1)
+    @test hasintercept(t) == false
 
     ## intercept-only
-    t = Terms(@formula(y ~ 1))
-    @test t.response == true
-    @test t.intercept == true
-    @test t.terms == []
-    @test t.eterms == [:y]
+    t = @formula(y ~ 1)
+    @test hasresponse(t) == true
+    @test hasintercept(t) == true
+    @test t.rhs == One
+    @test issetequal(terms(t), (One, y))
 
     ## terms add
-    t = Terms(@formula(y ~ 1 + x1 + x2))
-    @test t.intercept == true
-    @test t.terms == [:x1, :x2]
-    @test t.eterms == [:y, :x1, :x2]
+    t = @formula(y ~ 1 + x1 + x2)
+    @test hasintercept(t) == true
+    @test t.rhs == (One, x1, x2)
+    @test issetequal(terms(t), [y, One, x1, x2])
 
-    ## implicit intercept behavior:
-    t = Terms(@formula(y ~ x1 + x2))
-    @test t.intercept == true
-    @test t.terms == [:x1, :x2]
-    @test t.eterms == [:y, :x1, :x2]
+    ## implicit intercept behavior: NO intercept after @formula
+    t = @formula(y ~ x1 + x2)
+    @test hasintercept(t) == false
+    @test t.rhs == (x1, x2)
+    @test issetequal(terms(t), [y, x1, x2])
 
     ## no intercept
-    t = Terms(@formula(y ~ 0 + x1 + x2))
-    @test t.intercept == false
-    @test t.terms == [:x1, :x2]
+    t = @formula(y ~ 0 + x1 + x2)
+    @test hasintercept(t) == false
+    @test t.rhs == term(0, :x1, :x2)
 
-    @test t ==
-        Terms(@formula(y ~ -1 + x1 + x2)) ==
-        Terms(@formula(y ~ x1 - 1 + x2)) ==
-        Terms(@formula(y ~ x1 + x2 -1))
+    # subtraction not supported:
+    @test_skip @formula(y ~ -1 + x1 + x2) ==
```

Comment:
Should this be `@test_broken`?

Reply:
I think it errors or something, I remember there being a reason why I did this...